File: v1/agent.py
from __future__ import annotations

from pathlib import Path

from lovia import Agent
from pydantic import BaseModel

from common.models import deepseek_model


class GeneratedTest(BaseModel):
    code: str
    notes: str


def make_agent() -> Agent:
    prompt = (Path(__file__).parent / "prompts/generate.md").read_text(encoding="utf-8")
    return Agent(
        name="test-generator-v1",
        instructions=prompt,
        model=deepseek_model(),
        output_type=GeneratedTest,
    )

File: src/v1/agent.ts
/**
 * v1 单次生成 Agent：据静态快照一次性写出用例（结构化输出 GeneratedTest）。
 */

import { readFileSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';

import { generateObject } from 'ai';
import { z } from 'zod';

import { deepseekModel } from '../common/models';

export const GeneratedTest = z.object({
  code: z.string().describe('完整的 TypeScript @playwright/test 用例，纯代码，不含 markdown fence'),
  notes: z.string().describe('selector 选择策略与已知限制'),
});
export type GeneratedTest = z.infer<typeof GeneratedTest>;

const PROMPT = readFileSync(
  join(dirname(fileURLToPath(import.meta.url)), 'prompts', 'generate.md'),
  'utf-8',
);

export async function generateTest(userPrompt: string): Promise<GeneratedTest> {
  const { object } = await generateObject({
    model: deepseekModel(),
    schema: GeneratedTest,
    system: PROMPT,
    prompt: userPrompt,
  });
  return object;
}
